src/TestSuite/Fixture/TruncationStrategy.php

```diff
@@ -0,0 +1,126 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\TestSuite\Fixture;
+
+use Cake\Database\Connection;
+use Cake\Database\Schema\SqlGeneratorInterface;
+use Cake\Datasource\ConnectionManager;
+
+/**
+ * Fixture state strategy that truncates tables before a test.
+ *
+ * This strategy is slower than the TransactionStrategy, but
+ * allows tests to reset state when applications require operations
+ * that cannot be executed in a transaction. An example
+ * of this is DDL operations in MySQL which auto-commit any open
+ * transactions.
+ *
+ * This mode also offers 'backwards compatible' behavior
+ * with the schema + data fixture system.
+ */
+class TruncationStrategy
+{
+    /**
+     * Constructor.
+     *
+     * @param bool $enableLogging Whether or not to enable query logging.
+     * @return void
+     */
+    public function __construct(bool $enableLogging = false)
+    {
+        $this->aliasConnections($enableLogging);
+    }
+
+    /**
+     * Alias non test connections to the test ones
+     * so that models reach the test database connections instead.
+     *
+     * @param bool $enableLogging Whether or not to enable query logging.
+     * @return void
+     */
+    protected function aliasConnections(bool $enableLogging): void
```

Comment:
This code is somewhat duplicate but not entirely duplicate from the TransactionStrategy. I would like to clean up the duplication once I have a better handle on what the abstractions should be.

Reply:
This piece of code is very useful, and will be needed when the Schema is being created, for example before the migrations get run.

Therefore I suggest to move it in a dedicated class, so it can be called by other instances.

My proposal would be `Cake\TestSuite\TestConnectionManager.php`. I will include that in my coming PR for the Schema management.

Reply:
Separating it out makes sense to me.